libp2p/switch.nim

```diff
@@ -167,13 +167,9 @@ proc mux(s: Switch, conn: Connection): Future[Muxer] {.async, gcsafe.} =
   muxer.streamHandler = s.streamHandler
 
   s.connManager.storeOutgoing(conn)
-  s.connManager.storeMuxer(muxer)
-
-  # start muxer read loop - the future will complete when loop ends
-  let handlerFut = muxer.handle()
```

Comment:
afair, this was done so that the muxer would be available in connmanager for `handle` handlers that might run _before_ `storeMuxer`

Reply:
Yes this is suspect as well and I did consider this when making the change. It might be the change that broke this PR, but I have not yet been able to reproduce it. Any chance you can try reverting this and see if it resolves the potential issues you were seeing on our infra?